### PR TITLE
ref(checker): Accept CheckConfig in checker

### PR DESCRIPTION
**Pull request by evanpurkhiser**
We'll need tihs once we start firing off requests from the ConfigStore

### CODE REVIEW

File: src/types/check_config.rs [hunk 1]
```diff
@@ -65,16 +65,22 @@ mod tests {
 
     use super::{CheckConfig, CheckInterval};
 
+    impl Default for CheckConfig {
```
Comment (evanpurkhiser) [inline src/types/check_config.rs:68]: Moved the Default implemented for testing here

File: src/checker.rs [hunk 2]
```diff
@@ -1,44 +1,38 @@
...
 async fn do_request(
     client: &Client,
-    url: &str,
+    check_config: &CheckConfig,
     sentry_trace: &str,
 ) -> (RequestType, Result<Response, reqwest::Error>) {
     let head_response = match client
-        .head(url)
+        .head(check_config.url.as_str())
+        .timeout(check_config.timeout)
```
Comment (evanpurkhiser) [inline src/checker.rs:35]: Timeout read from config

File: src/producer.rs [hunk 3]
```diff
@@ -58,20 +58,19 @@ impl ResultProducer {
...
     use sentry::protocol::{SpanId, TraceId};
-    use uuid::Uuid;
+    use uuid::{uuid, Uuid};
 
     pub async fn send_result(result: CheckStatus) -> Result<(), ExtractCodeError> {
         let guid = Uuid::new_v4();
         let result = CheckResult {
             guid,
-            subscription_id: SUBSCRIPTION_ID,
+            subscription_id: uuid!("23d6048d67c948d9a19c0b47979e9a03"),
```
Comment (evanpurkhiser) [inline src/producer.rs:73]: using dummy data

File: src/scheduler.rs [hunk 4]
```diff
@@ -25,9 +28,14 @@ pub async fn run_scheduler(config: &Config) -> Result<(), JobSchedulerError> {
...
 
-            let check_result = job_checker
-                .check_url("https://downtime-simulator-test1.vercel.app")
-                .await;
+            let test_config = CheckConfig {
+                url: "https://downtime-simulator-test1.vercel.app".to_string(),
+                subscription_id: uuid!("663399a09e6340a79c3c7a3f26878904"),
+                interval: CheckInterval::OneMinute,
+                timeout: Duration::from_secs(5),
+            };
```
Comment (evanpurkhiser) [inline src/scheduler.rs:36]: The entire example config we have lives here now.

File: src/checker.rs [hunk 5]
```diff
@@ -1,44 +1,38 @@
...
 
-pub struct CheckerConfig {
-    /// How long will we wait before we consider the request to have timed out.
-    pub timeout: Duration,
-}
-
-impl Default for CheckerConfig {
-    fn default() -> Self {
-        Self {
-            timeout: Duration::from_secs(5),
```
Comment (evanpurkhiser) [inline src/checker.rs:19]: Duration is part of the individual check configs now

File: src/checker.rs [hunk 6]
```diff
@@ -78,9 +73,8 @@ fn dns_error(err: &reqwest::Error) -> Option<String> {
 }
 
 impl Checker {
-    pub fn new(config: CheckerConfig) -> Self {
+    pub fn new(_config: CheckerConfig) -> Self {
```
Comment (wedamija) [inline src/checker.rs:76]: Do we still need to pass the config in here now?

**Review by wedamija — APPROVED** [review]

File: src/checker.rs (hunk 6 above)
Comment (evanpurkhiser) [reply src/checker.rs:76]: just left it for now, maybe I'll just remove it all together after this PR

**Pull request merged**